open-x-htb.js

```diff
@@ -104,65 +105,79 @@ function OpenXHtb(configs) {
-      var gdprPrivacyEnabled = ComplianceService.isPrivacyEnabled();
-
-
-
-      for (var i = 0; i < returnParcels.length; i++) {
-        auidString += returnParcels[i].xSlotRef.adUnitId.toString() + ',';
-        ausString += Size.arrayToString(returnParcels[i].xSlotRef.sizes, ',') + '|';
-      }
-
-      auidString = auidString.slice(0, -1);
-      ausString = ausString.slice(0, -1);
-
-      var queryObj = {
-        auid: auidString,
-        aus: ausString,
-        ju: Browser.getPageUrl(),
-        jr: Browser.getReferrer(),
-        ch: configs.charset,
-        tz: System.getTimezoneOffset(),
-        bc: 'hb_ix',
-        be: 1,
-        res: Size.arrayToString([
-          [Browser.getScreenWidth(), Browser.getScreenHeight()]
-        ]),
-        tws: Size.arrayToString([
-          [Browser.getViewportWidth(), Browser.getViewportHeight()]
-        ]),
-        ifr: Browser.isTopFrame() ? 0 : 1,
-        callback: 'window.' + SpaceCamp.NAMESPACE + '.OpenXHtb.adResponseCallbacks.' + callbackId,
-        cache: new Date().getTime()
-      };
-
-      if (gdprPrivacyEnabled) {
-        if (gdprConsent.consentString !== void(0)) {
-          queryObj.gdpr_consent = gdprConsent.consentString;
+        var gdprConsent = ComplianceService.gdpr.getConsent();
+        var gdprPrivacyEnabled = ComplianceService.isPrivacyEnabled();
+
+        var tradeDeskId;
+
+        for (var i = 0; i < returnParcels.length; i++) {
+            auidString += returnParcels[i].xSlotRef.adUnitId.toString() + ',';
+            ausString += Size.arrayToString(returnParcels[i].xSlotRef.sizes, ',') + '|';
+        }
+
+        auidString = auidString.slice(0, -1);
+        ausString = ausString.slice(0, -1);
+
+        var identityData = returnParcels[0] && returnParcels[0].identityData;
+
+        if (identityData && identityData.AdserverOrgIp && identityData.AdserverOrgIp.data) {
+            var adsrvrUids = identityData.AdserverOrgIp.data.uids;
+            if (Utilities.isArray(adsrvrUids)) {
+                for (var j = 0; j < adsrvrUids.length; j++) {
+                    if (adsrvrUids[j].ext && adsrvrUids[j].ext.rtiPartner === 'TDID') {
+                        tradeDeskId = adsrvrUids[j].id;
+                    }
+                }
+            }
+        }
+
+        var queryObj = {
+            auid: auidString,
+            aus: ausString,
+            ju: Browser.getPageUrl(),
+            jr: Browser.getReferrer(),
+            ch: configs.charset,
+            tz: System.getTimezoneOffset(),
+            bc: 'hb_ix' + '_' + __profile.version,
+            be: 1,
+            res: Size.arrayToString([[Browser.getScreenWidth(), Browser.getScreenHeight()]]),
+            tws: Size.arrayToString([[Browser.getViewportWidth(), Browser.getViewportHeight()]]),
+            ifr: Browser.isTopFrame() ? 0 : 1,
+            callback: 'window.' + SpaceCamp.NAMESPACE + '.OpenXHtb.adResponseCallbacks.' + callbackId,
+            cache: new Date()
```

Comment:
You can make use of the `System.now()` function outlined in the README.md. It implements the same functionality.